core/modules/entity/entity.module

```diff
@@ -20,6 +20,406 @@ function entity_modules_disabled() {
 }
 
 /**
+ * Implements hook_permission().
+ */
+function entity_permission() {
+  return array(
+    'administer view modes' => array(
+      'title' => t('Add, edit and delete custom view modes.'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function entity_menu() {
+  $items['admin/structure/view-modes'] = array(
+    'title' => 'View modes',
+    'description' => 'Create new custom view modes for content.',
+    'page callback' => 'entity_view_mode_list',
+    'access arguments' => array('administer view modes'),
+    'file' => 'entity.admin.inc',
+  );
+  $items['admin/structure/view-modes/add/%'] = array(
+    'title' => 'Add view mode',
+    'page callback' => 'backdrop_get_form',
+    'page arguments' => array('entity_view_mode_edit_form', 4),
+    'access arguments' => array('administer view modes'),
+    'file' => 'entity.admin.inc',
+  );
+  $items['admin/structure/view-modes/edit/%/%'] = array(
+    'title' => 'Edit view mode',
+    'page callback' => 'backdrop_get_form',
+    'page arguments' => array('entity_view_mode_edit_form', 4, 5),
+    'access arguments' => array('administer view modes'),
+    'file' => 'entity.admin.inc',
+  );
+  $items['admin/structure/view-modes/delete/%/%'] = array(
+    'title' => 'Edit view mode',
+    'page callback' => 'backdrop_get_form',
+    'page arguments' => array('entity_view_mode_delete_form', 4, 5),
+    'access arguments' => array('administer view modes'),
+    'file' => 'entity.admin.inc',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function entity_menu_local_tasks_alter(&$data, $router_item, $root_path) {
+  // On entity "Manage display" pages, $root_path is in the format
+  // 'admin/structure/types/manage/%/display' whereas entity admin paths are in
+  // the format'admin/structure/types/manage/%node_type'. To determine if this
+  // is a display page for an entity we loop through all available entities and
+  // replace the load argument in the admin paths with a '%' and append
+  // '/display' and check if this matches the current $root_path.
```

Comment:
Yeah, I agree. The add view mode link here shouldn't really be an action link anyway. My last push should have the links in the fieldset now, where the other view modes are listed (and where people are likely to look for that link anyway)